### PR TITLE
[User] rename and better use the singleton and factory patterns

### PR DESCRIPTION
**Pull request by ridz1208**
This removes the singleton logic from the User and adds a deprecation warning for the factory function. new explicitly named factories were added.

The user class should start using the ID field instead of userID (username) as primary key, this Pr contains some changes to help with that.

### CODE REVIEW

File: php/libraries/NDB_Client.class.inc [hunk 1]
```diff
@@ -89,7 +89,7 @@ class NDB_Client
         if (!$this->_isWebClient) {
             // Set user as unix username.
             // Requires Loris to have user with this username
-            $user =& User::singleton(getenv('USER'));
+            $user = \User::singleton(getenv('USER'));
```
Comment (ridz1208) [inline php/libraries/NDB_Client.class.inc]: to be replaced

File: tools/fix_timepoint_date_problems.php [hunk 2]
```diff
@@ -249,7 +249,7 @@
 function addInstrument($sessionID, $testName)
 {
     // check the user $_ENV['USER']
-    $user =& User::singleton(getenv('USER'));
+    $user = \User::singleton(getenv('USER'));
```
Comment (ridz1208) [inline tools/fix_timepoint_date_problems.php]: replace this

File: tools/fix_timepoint_date_problems.php [hunk 3]
```diff
@@ -329,7 +329,7 @@ function addInstrument($sessionID, $testName)
 function fixDate($candID, $dateType, $newDate, $sessionID=null)
 {
     // check the user $_ENV['USER']
-    $user =& User::singleton(getenv('USER'));
+    $user = \User::singleton(getenv('USER'));
```
Comment (ridz1208) [inline tools/fix_timepoint_date_problems.php]: replace this

**Comment by driusan** [comment]
This doesn't seem to remove the singleton pattern, it just renames it (which is still a step forward..)

**Comment by ridz1208** [comment]
to be finalized

**Pull request closed without merging**